contracts/ColonyExpenditure.sol

```diff
@@ -0,0 +1,196 @@
+
+    emit ExpenditureTransferred(_id, _newOwner);
+  }
+
+  function cancelExpenditure(uint256 _id)
+    public
+    stoppable
+    expenditureExists(_id)
+    expenditureActive(_id)
+    expenditureOnlyOwner(_id)
+  {
+    expenditures[_id].status = ExpenditureStatus.Cancelled;
+
+    emit ExpenditureCancelled(_id);
+  }
+
+  function finalizeExpenditure(uint256 _id)
+    public
+    stoppable
+    expenditureExists(_id)
+    expenditureActive(_id)
+    expenditureOnlyOwner(_id)
+  {
+    FundingPot storage fundingPot = fundingPots[expenditures[_id].fundingPotId];
+    require(fundingPot.payoutsWeCannotMake == 0, "colony-expenditure-not-funded");
+
+    expenditures[_id].status = ExpenditureStatus.Finalized;
+    expenditures[_id].finalizedTimestamp = now;
+
+    emit ExpenditureFinalized(_id);
+  }
+
+  function setExpenditureRecipient(uint256 _id, uint256 _slot, address payable _recipient)
+    public
+    stoppable
+    expenditureExists(_id)
+    expenditureActive(_id)
+    expenditureOnlyOwner(_id)
+  {
+    expenditureSlots[_id][_slot].recipient = _recipient;
+
+    emit ExpenditureRecipientSet(_id, _slot, _recipient);
+  }
+
+  function setExpenditureSkill(uint256 _id, uint256 _slot, uint256 _skillId)
+    public
+    stoppable
+    expenditureExists(_id)
+    expenditureActive(_id)
+    expenditureOnlyOwner(_id)
+    skillExists(_skillId)
+    validGlobalSkill(_skillId)
+  {
+    // We only allow setting of the first skill here.
+    // If we allow more in the future, make sure to have a hard limit that
+    // comfortably limits respondToChallenge's gas.
+    expenditureSlots[_id][_slot].skills = new uint256[](1);
+    expenditureSlots[_id][_slot].skills[0] = _skillId;
+
+    emit ExpenditureSkillSet(_id, _slot, _skillId);
+  }
+
+  function setExpenditurePayoutScalar(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _id,
+    uint256 _slot,
+    uint256 _payoutScalar
+  )
+    public
+    stoppable
+    authDomain(_permissionDomainId, _childSkillIndex, expenditures[_id].domainId)
+  {
+    require(_payoutScalar <= MAX_PAYOUT_SCALAR, "colony-expenditure-payout-scalar-too-large");
+
+    expenditureSlots[_id][_slot].payoutScalar = packPayoutScalar(_payoutScalar);
+  }
+
+  function setExpenditureClaimDelay(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _id,
+    uint256 _slot,
+    uint256 _claimDelay
+  )
+    public
+    stoppable
+    authDomain(_permissionDomainId, _childSkillIndex, expenditures[_id].domainId)
+  {
+    expenditureSlots[_id][_slot].claimDelay = _claimDelay;
+  }
+
+  // Public view functions
+
+  function getExpenditureCount() public view returns (uint256) {
+    return expenditureCount;
+  }
+
+  function getExpenditure(uint256 _id) public view returns (Expenditure memory expenditure) {
+    expenditure = expenditures[_id];
+  }
+
+  function getExpenditureRecipient(uint256 _id, uint256 _slot) public view returns (address) {
```

Comment:
Ultimately it boils down to ease of passing data around between contracts. Since mappings don't play nicely with `ABIEncoderV2`, returning structs-with-mappings requires unpacking them into a tuple, i.e. `returns(uint256, uint256)`, which I know see as an anti-pattern because it necessitates code on the other side which looks like this:

```
uint256 field;
(, field) = contract.foo();
```
I find that pattern to be cumbersome and difficult to use, and individual getters mean that we can access these data from other contracts with a much friendlier syntax:

```
uint256 field = contract.foo()
```

The reason why the `ExpenditureSlot` was separated from the `Expenditure` was exactly to remove all mappings from the `Expenditure`, and allow it to be returned directly as a struct.

Reply:
Thoughts on this? ^^

Reply:
I mean, that's how we implemented tasks. But do you not think that for consistency, if you're taking this approach, everything should be in the same struct and then everything should have an individual getter? It is weird that for some properties of an expenditure I can use the dot notation, and for others I need to use a separate getter.

I'm still not convinced I like it though.

Reply:
I think my preference would actually be to pull out the non-mapping parts of `ExpenditureSlot` into a separate struct, which would let us return `ExpenditureSlot` (sans payouts) directly. That way there's more consistency with the expenditure and the only stand-alone getter is for the payment, which would need to be separate anyway. I agree that individual getters are clunky and add cruft to the contract.

Reply:
Yeah, I'd be happier with that, it's straddling two philosophies at the moment.